### ISSUE TITLE
make deps requires haskell dependencies by default

### ISSUE THREAD

**Issue by bl4ck5un** [kframework/evm-semantics#324]
Following the README, one would expect `make deps` doesn't compile the optional haskell-backend or require those dependencies (e.g., `stack`). However `make deps` gives the following error:

~~~
main:
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] K Framework Tool Parent ............................ SUCCESS [ 19.730 s]
[INFO] K Framework KORE ................................... SUCCESS [01:07 min]
[INFO] K Framework Tool Kernel ............................ SUCCESS [ 35.062 s]
[INFO] K Framework KTree .................................. SUCCESS [  0.061 s]
[INFO] K Framework Ocaml Backend .......................... SUCCESS [  0.343 s]
[INFO] K Framework Java Backend ........................... SUCCESS [ 19.760 s]
[INFO] K Framework Haskell Backend ........................ FAILURE [  5.224 s]
[INFO] K Framework LLVM Backend ........................... SKIPPED
[INFO] K Framework Tool Distribution ...................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 03:16 min
[INFO] Finished at: 2019-03-05T14:10:11-05:00
[INFO] Final Memory: 62M/538M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.7:run (build-haskell) on project haskell-backend: An Ant BuildException has occured: Execute failed: java.io.IOException: Cannot run program "stack" (in directory "/home/fanz/tmp/evm-semantics/.build/k/haskell-backend/src/main/native/haskell-backend"): error=2, No such file or directory
[ERROR] around Ant part ...<exec failonerror="true" dir="/home/fanz/tmp/evm-semantics/.build/k/haskell-backend/src/main/native/haskell-backend" executable="stack">... @ 4:139 in /home/fanz/tmp/evm-semantics/.build/k/haskell-backend/target/antrun/build-main.xml
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :haskell-backend
~~~

It's easy to work this around (following the instructions for building haskell backend) but just want to point this out.

**Comment by ehildenb** [comment]
@bl4ck5un please check #325.

**Comment by ehildenb** [comment]
Addressed by #325.
